crates/types/src/camera_matrix.rs

```diff
@@ -89,12 +90,118 @@ impl CameraMatrix {
             2.0 * (image_dim * 0.5 / focal_length).atan()
         })
     }
+
+    pub fn into_corrected(
+        self,
+        correction_in_robot: Rotation3<f32>,
+        correction_in_camera: Rotation3<f32>,
+    ) -> Self {
+        let camera_to_head = self.camera_to_head;
+        let robot_to_head = self.camera_to_head * self.robot_to_camera;
+        let head_to_robot = robot_to_head.inverse();
+        let ground_to_robot = self.camera_to_robot * self.ground_to_camera;
+        let robot_to_ground = ground_to_robot.inverse();
+
+        let corrected_camera_to_head =
+            Isometry3::from_parts(Default::default(), correction_in_camera.inverse().into())
+                * camera_to_head;
+        let corrected_head_to_camera = corrected_camera_to_head.inverse();
+        let corrected_head_to_robot =
+            Isometry3::from_parts(Default::default(), correction_in_robot.inverse().into())
+                * head_to_robot;
```

Comment:
I may be wrong, but I think this multiplication are the wrong way around.
In the first calculation, when multiplying a vector from the right, it is first transformed from camera coordinates to head coordinates before being rotated "in camera" even though it is already in head coordinates.
The `corrected_head_to_robot` calculation seems correct, since you want to rotate in the destination frame of that transformation anyway.

Reply:
@jonathan-hellwig Could you comment on this? :joy:

Reply:
For the first one I agree with @knoellle and for the second one I would suggest a different name. I'll put it into math notation to make it clear: Let $C$ be the uncorrected camera system, $\tilde{C}$ the corrected camera system and $H$ be the head system. Then, the first transformation is given by $T_{\tilde{C}}^H = T^H_C T^C_{\tilde{C}}$. 
For the second transformation we have the uncorrected robot system $R$, the corrected robot system $\tilde{R}$ and the head system $H$. Then, the second transformation is given by $T^{\tilde{R}}_H = T^{\tilde{R}}_R T^R_H$. 
I would suggest to name the transformation `head_to_corrected_robot`, because it makes the reference frame of a point transformed by $T^{\tilde{R}}_H$ more precise. If you just use `corrected_*` everytime it is not clear whether you modified the original or the target coordinate sytem.

Reply:
Thanks, I renamed some variables and swapped the multiplication operands in the first calculation.